### PR TITLE
Add remove repo

### LINKED ISSUE #2650: Build: Remove GitHub repo for a snap - frontend

### PR DESCRIPTION
**Pull request by solazio**
## Done

- Add disconnect repo

## Issue / Card

Fixes #2650 

## QA

- Pull the branch
- Run the site using the command `./run`
- View the site locally in your web browser at: http://0.0.0.0:8004/<snap_name>/builds
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Go to a snap with builds and click on `Disconnect repo` link and then click `Confirm` button
- See the repo has been disconnected
- Perhaps this should be QAed from different GitHub accounts as well.

### CODE REVIEW

File: webapp/publisher/snaps/build_views.py [hunk 1]
```diff
@@ -364,7 +364,9 @@ def get_disconnect_repo(snap_name):
     except ApiError as api_error:
         return _handle_error(api_error)
 
-    return flask.render_template("publisher/disconnect-repo.html")
+    return flask.render_template(
+        "publisher/disconnect-repo.html", snap_name=snap_name
```
Comment (jkfran) [inline webapp/publisher/snaps/build_views.py:368]: :+1:  haha

File: templates/publisher/_repo-disconnect-modal.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<div class="p-modal" data-js="repo-disconnect-modal" id="repo-disconnect-modal" style="display: none;">
+  <div class="p-modal__dialog" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description">
+    <header class="p-modal__header">
+      <h4><i class="p-icon--warning"></i>&nbsp;&nbsp;Disconnecting your repository</h4>
+      <button class="p-modal__close" aria-label="Close active modal" aria-controls="repo-disconnect-modal">Close</button>
+    </header>
+    <p>By disconnecting this repository from your snap, the following will occur:</p>
+    <ul>
+      <li>The previous builds you did on the snap may disappear from the Builds page</li>
```
Comment (Lukewh) [inline templates/publisher/_repo-disconnect-modal.html:9]:
```suggestion
      <li>Previous builds for this snap will disappear from the Builds page</li>
```

File: templates/publisher/_repo-disconnect-modal.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      <li>You will be able to continue to release your previous revisions on the Releases tab</li>
+    </ul>
+    <p>Do you wish to <strong>continue disconnecting your repository?</strong></p>
+    <form class="p-form" method="POST" action="/{{ snap_name }}/builds/disconnect">
+      <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
+      <p class="u-align--right">
+        <button aria-label="Close active modal" aria-controls="repo-disconnect-modal"class="p-button--neutral">Cancel</button>
+        <button type="submit" class="p-button--positive">Confirm</button>
```
Comment (Lukewh) [inline templates/publisher/_repo-disconnect-modal.html:17]: Can we do an immediate disable with a delayed spinner on this button - for me I clicked it and it felt like nothing was happening for a few seconds and I was tempted to click it again.

File: templates/publisher/_repo-disconnect-modal.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      <li>Previous builds for this snap will disappear from the Builds page</li>
+      <li>You will be able to continue to release your previous revisions on the Releases tab</li>
+    </ul>
+    <p>Do you wish to <strong>continue disconnecting your repository?</strong></p>
+    <form class="p-form" method="POST" action="/{{ snap_name }}/builds/disconnect" name="repoDisconnectForm">
+      <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
+      <p class="u-align--right">
+        <button aria-label="Close active modal" aria-controls="repo-disconnect-modal"class="p-button--neutral">Cancel</button>
+        <button class="p-button--positive p-button-spinner" data-js="repo-disconnect-confirm">
+          <span class="p-button__spinner vanilla-workaround--dark">
```
Comment (Lukewh) [inline templates/publisher/_repo-disconnect-modal.html:18]: the spinner isn't showing for me

Comment (solazio) [reply templates/publisher/_repo-disconnect-modal.html:18]: honestly I don't know how to fix that. It seems it is related to [the form submit](https://github.com/canonical-web-and-design/snapcraft.io/pull/2654/files#diff-c4e1a01079db29511ae3baf4259a059dR39) - if I don't submit the form, it shows the spinner. I tried to submit the for after 500ms, but it doesn't help... don't know what else to do. Anyway, it works fine on chrome.

**Comment by solazio** [comment]
we have a [new issue](https://github.com/canonical-web-and-design/snapcraft.io/issues/2659) for the missing spinner in Firefox

**Pull request merged**